diff -Naur config4cpp/Makefile.inc config4cpp.now/Makefile.inc
--- config4cpp/Makefile.inc	2012-02-02 04:47:18.000000000 -0600
+++ config4cpp.now/Makefile.inc	2020-03-08 17:06:30.659050018 -0500
@@ -16,10 +16,10 @@
 INC_DIR=$(PROJECT_ROOT)/include
 
 ifeq ($(BUILD_TYPE),debug)
-LIB_CONFIG4CPP=libconfig4cppd.a
+LIB_CONFIG4CPP=libconfig4cppd.so
 LINK_WITH_CONFIG4CPP=-L$(LIB_DIR) -lconfig4cppd
 else
-LIB_CONFIG4CPP=libconfig4cpp.a
+LIB_CONFIG4CPP=libconfig4cpp.so
 LINK_WITH_CONFIG4CPP=-L$(LIB_DIR) -lconfig4cpp
 endif
 
diff -Naur config4cpp/src/Makefile config4cpp.now/src/Makefile
--- config4cpp/src/Makefile	2011-12-22 13:48:36.000000000 -0600
+++ config4cpp.now/src/Makefile	2020-03-08 17:21:26.947105753 -0500
@@ -99,8 +99,7 @@
 all:	config4cpp$(EXE_EXT) config2cpp$(EXE_EXT) config2cpp-nocheck$(EXE_EXT)
 
 $(LIB_CONFIG4CPP):	$(LIB_OBJ_FILES)
-	$(AR) $(AR_FLAGS) $(LIB_CONFIG4CPP) $(LIB_OBJ_FILES)
-	$(RANLIB) $(RANLIB_FLAGS) $(LIB_CONFIG4CPP)
+	$(CXX) $(CXX_FLAGS) -Wl,-soname,$(LIB_CONFIG4CPP) -shared -fPIC $^ -o $@
 	cp $(LIB_CONFIG4CPP) $(LIB_DIR)
 
 DefaultSecurity.cpp DefaultSecurity.h: DefaultSecurity.cfg config2cpp-nocheck$(EXE_EXT)
@@ -113,7 +112,7 @@
 
 config2cpp$(EXE_EXT):	config2cpp-main.o Config2Cpp.o $(LIB_CONFIG4CPP)
 		$(CXX) -o config2cpp$(EXE_EXT) config2cpp-main.o Config2Cpp.o \
-			$(LIB_CONFIG4CPP)
+			-L. -lconfig4cpp
 		cp config2cpp$(EXE_EXT) $(BIN_DIR)
 
 config2cpp-nocheck$(EXE_EXT):	config2cpp-nocheck-main.o Config2Cpp.o
@@ -121,7 +120,7 @@
 			config2cpp-nocheck-main.o Config2Cpp.o
 
 config4cpp$(EXE_EXT):	config4cpp.o $(LIB_CONFIG4CPP)
-		$(CXX) -o config4cpp$(EXE_EXT) config4cpp.o $(LIB_CONFIG4CPP)
+		$(CXX) -o config4cpp$(EXE_EXT) config4cpp.o -L. -lconfig4cpp
 		cp config4cpp$(EXE_EXT) $(BIN_DIR)
 
 config4cpp.o:	config4cpp.cpp 	$(LIB_HDR_FILES)
