From d8a7c1bee135e21db569d6935d0d86329e0f8fc0 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Tue, 27 Sep 2022 17:20:56 -0400
Subject: [PATCH 2/6] pillow png optimizer: use sane default options, 256 color
 + dithering

---
 overviewer_core/optimizeimages.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/overviewer_core/optimizeimages.py b/overviewer_core/optimizeimages.py
index ae42c35..cb92529 100644
--- a/overviewer_core/optimizeimages.py
+++ b/overviewer_core/optimizeimages.py
@@ -212,7 +212,7 @@ class oxipng(Optimizer, PNGOptimizer):
 
 class pillowpng(Optimizer, PNGOptimizer):
 
-    def __init__(self, colors=20, dither=False):
+    def __init__(self, colors=256, dither=True):
         if colors < 2:
             raise Exception("You can't use less than 2 colors")
         self.colors = colors
-- 
2.50.1

