From e4ad18f03a927453ef337e9b6601de1b4c01b33d Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Fri, 18 Jul 2025 13:00:35 -0400
Subject: [PATCH 6/6] build: remove stupid Werror on
 declaration-after-statement

---
 setup.py | 2 --
 1 file changed, 2 deletions(-)

diff --git a/setup.py b/setup.py
index 6020d62..47ab8a7 100755
--- a/setup.py
+++ b/setup.py
@@ -322,8 +322,6 @@ class CustomBuildExt(build_ext):
             for e in self.extensions:
                 e.extra_compile_args.append("-Wno-unused-variable") # quell some annoying warnings
                 e.extra_compile_args.append("-Wno-unused-function") # quell some annoying warnings
-                e.extra_compile_args.append("-Wdeclaration-after-statement")
-                e.extra_compile_args.append("-Werror=declaration-after-statement")
                 e.extra_compile_args.append("-O3")
                 e.extra_compile_args.append("-std=gnu99")
 
-- 
2.50.1

