From 0af082ce383d2bdb4f65e2962fbb9b55005a6c36 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Fri, 20 Feb 2026 00:55:32 -0500
Subject: [PATCH] remove use of imp - imp is no longer supported

---
 overviewer_core/util.py | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/overviewer_core/util.py b/overviewer_core/util.py
index 403cbcd..0318cee 100644
--- a/overviewer_core/util.py
+++ b/overviewer_core/util.py
@@ -18,7 +18,6 @@ Misc utility routines used by multiple files that don't belong anywhere else
 """
 
 import errno
-import imp
 import os.path
 import platform
 import sys
@@ -28,7 +27,7 @@ from subprocess import PIPE, Popen
 
 
 def get_program_path():
-    if hasattr(sys, "frozen") or imp.is_frozen("__main__"):
+    if hasattr(sys, "frozen"):
         return os.path.dirname(sys.executable)
     else:
         try:
-- 
2.52.0

