From 34d062f69f48423ab2a8b90ac2587eede276350f Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Tue, 1 Dec 2020 22:43:42 -0600
Subject: [PATCH 9002/9002] blk: request hook points

---
 block/blk-core.c | 8 ++++++++
 block/blk-mq.c   | 5 +++++
 2 files changed, 13 insertions(+)

diff --git a/block/blk-core.c b/block/blk-core.c
index 0014e7caae3d..47ad05192e4e 100644
--- a/block/blk-core.c
+++ b/block/blk-core.c
@@ -45,6 +45,10 @@
 #define CREATE_TRACE_POINTS
 #include <trace/events/block.h>
 
+/* EXPLODE */
+#include <linux/explode.h>
+/* END EXPLODE */
+
 #include "blk.h"
 #include "blk-mq.h"
 #include "blk-mq-sched.h"
@@ -1131,6 +1135,10 @@ static blk_qc_t __submit_bio_noacct(struct bio *bio)
 		bio_list_on_stack[1] = bio_list_on_stack[0];
 		bio_list_init(&bio_list_on_stack[0]);
 
+		/* EXPLODE */
+		EXPLODE_HOOK(bio_start_write, bio);
+		/* END EXPLODE */
+
 		ret = __submit_bio(bio);
 
 		/*
diff --git a/block/blk-mq.c b/block/blk-mq.c
index ca2fdb58e7af..c3309e441b22 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -536,6 +536,11 @@ inline void __blk_mq_end_request(struct request *rq, blk_status_t error)
 {
 	u64 now = 0;
 
+	/* EXPLODE */
+	if(error == BLK_STS_OK)
+		EXPLODE_HOOK(req_write_done, rq);
+	/* END EXPLODE */
+
 	if (blk_mq_need_time_stamp(rq))
 		now = ktime_get_ns();
 
-- 
2.28.0

