From 09cbb0a544f92adccb09c8106dad088eb4f366e5 Mon Sep 17 00:00:00 2001
From: Tianhao Chai <cth451@gmail.com>
Date: Fri, 12 Aug 2022 01:15:56 -0400
Subject: [PATCH] meta: fix building on arch w/target clone but not added yet

---
 src/libminemap.h | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/src/libminemap.h b/src/libminemap.h
index c91e358..28b5972 100644
--- a/src/libminemap.h
+++ b/src/libminemap.h
@@ -16,9 +16,12 @@
 
 #if __has_attribute(target_clones) && !defined(_WIN32)
 # if (defined(__amd64__) || defined(__x86_64__))
-# define __MINEMAP_CLONE__ __attribute((target_clones("default", "sse2", "sse3", "sse4.1", "sse4.2", "popcnt", "avx", "avx2")))
+#  define __MINEMAP_CLONE__ __attribute((target_clones("default", "sse2", "sse3", "sse4.1", "sse4.2", "popcnt", "avx", "avx2")))
 # elif (defined(__PPC64__))
-# define __MINEMAP_CLONE__ __attribute__((target_clones("default","cpu=power9")))
+#  define __MINEMAP_CLONE__ __attribute__((target_clones("default","cpu=power9")))
+# else
+#  pragma message( "Warning: Target clone not supported yet" )
+#  define __MINEMAP_CLONE__
 # endif
 #else
 #pragma message( "Warning: No target clone support" )
-- 
2.37.1

